addons/google_calendar/models/calendar.py

```diff
@@ -109,22 +109,29 @@ def _odoo_attendee_commands(self, google_event):
         if google_event.exists(self.env):
             existing_attendees = self.browse(google_event.odoo_id(self.env)).attendee_ids
         attendees_by_emails = {tools.email_normalize(a.email): a for a in existing_attendees}
-        for attendee in google_attendees:
-            email = attendee.get('email')
-
+        partners = self.env['mail.thread']._mail_find_partner_from_emails(emails, records=self, force_create=True, extra_domain=[('type', '!=', 'private')])
+        for attendee in zip(emails, partners, google_attendees):
+            email = attendee[0]
```

Comment:
Question: the [15.0 fix](https://github.com/odoo/odoo/pull/82454/files) which was merged looks decidedly simpler. Did you have to include other changes that were done from 14.0->15.0?

If yes -> is it possible to cherry-pick these, or at least put them in a separate commit from the actual fix for this issue?

If no -> can you explain a bit how you arrived at the current code

Reply:
Yes, I had to  change [_odoo_attendee_commands()](https://github.com/OCA/OCB/blob/59cf062c23728ea568b80688db94215a5c73af19/addons/google_calendar/models/calendar.py#L97) method significantly, as it is basically different from the 15.0 version. The difference that required these changes is [how partners are retreived](https://github.com/OCA/OCB/blob/4d8d5dcdecc0beabec56d3150f583d771ab1e2d5/addons/google_calendar/models/calendar.py#L146). As you can see, cherry-picking won't work here.